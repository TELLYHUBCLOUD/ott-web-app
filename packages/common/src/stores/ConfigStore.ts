import { ACCESS_MODEL, OTT_GLOBAL_PLAYER_ID } from '../constants';
import type { AccessModel, Config, IntegrationType } from '../../types/config';
import type { Settings } from '../../types/settings';

import { createStore } from './utils';

type ConfigState = {
  loaded: boolean;
  config: Config;
  accessModel: AccessModel;
  settings: Settings;
  integrationType: IntegrationType | null;
  isSandbox: boolean;
  clientId: string | null;
  offers: string[];
};

export const useConfigStore = createStore<ConfigState>('ConfigStore', () => ({
  loaded: false,
  config: {
    id: '',
    siteName: '',
    description: '',
    player: '',
    assets: {},
    content: [],
    menu: [],
    integrations: {
      cleeng: {
        useSandbox: true,
      },
      jwp: {
        clientId: null,
        assetId: null,
        useSandbox: true,
      },
    },
    styling: {
      footerText: '',
    },
  },
  settings: {
    additionalAllowedConfigSources: [],
    playerId: OTT_GLOBAL_PLAYER_ID,
  },
  accessModel: ACCESS_MODEL.SVOD,
  integrationType: null,
  isSandbox: true,
  clientId: null,
  offers: [],
}));
