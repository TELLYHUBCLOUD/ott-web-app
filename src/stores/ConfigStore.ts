import { createStore } from './utils';

import type { AccessModel, Config } from '#types/Config';
import { API_HOST, DEFAULT_PLAYER } from '#src/config';

export enum PersonalShelf {
  ContinueWatching = 'continue_watching',
  Favorites = 'favorites',
}

export const PersonalShelves = [PersonalShelf.Favorites, PersonalShelf.ContinueWatching];

type CleengData = {
  cleengId: string | null | undefined;
  cleengSandbox: boolean;
  monthlyOfferId: string;
  yearlyOfferId: string;
};

type ConfigState = {
  config: Config;
  accessModel: AccessModel;
  apiHost: string;
  getPlayer: () => string;
  getCleengData: () => CleengData;
};

export const useConfigStore = createStore<ConfigState>('ConfigStore', (_, get) => ({
  config: {
    id: '',
    siteName: '',
    description: '',
    player: '',
    assets: {},
    content: [],
    menu: [],
    integrations: {
      cleeng: {
        useSandbox: true,
      },
      inplayer: {
        clientId: null,
        assetId: null,
        useSandbox: true,
      },
    },
    styling: {
      footerText: '',
    },
  },
  accessModel: 'SVOD',
  apiHost: import.meta.env.APP_API_BASE_URL || API_HOST.prd,
  getPlayer: (): string => {
    const host = get().apiHost;

    // For dev api host we use only one testing id, for prod it can be also set in .env file
    return host === API_HOST.dev ? DEFAULT_PLAYER.dev : import.meta.env.APP_DEFAULT_PLAYER ?? DEFAULT_PLAYER.prd;
  },
  getCleengData: (): CleengData => {
    const cleeng = get().config?.integrations?.cleeng;

    const cleengId = cleeng?.id;
    const cleengSandbox = !!cleeng?.useSandbox;
    const monthlyOfferId = cleeng?.monthlyOffer || '';
    const yearlyOfferId = cleeng?.yearlyOffer || '';

    return { cleengId, cleengSandbox, monthlyOfferId, yearlyOfferId };
  },
}));
