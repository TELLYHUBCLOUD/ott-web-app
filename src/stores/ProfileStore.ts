import type { Profile } from '#types/account';
import { createStore } from '#src/utils/createStore';

type ProfileStore = {
  profile: Profile | null;
  selectingProfileAvatar: string | null;
};

export const useProfileStore = createStore<ProfileStore>('AccountStore', () => ({
  profile: null,
  selectingProfileAvatar: null,
}));
