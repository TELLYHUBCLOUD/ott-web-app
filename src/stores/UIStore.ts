import type { Location } from 'react-router-dom';

import { createStore } from '#src/utils/createStore';

type UIState = {
  searchQuery: string;
  searchActive: boolean;
  userMenuOpen: boolean;
  languageMenuOpen: boolean;
  preSearchPage?: Location;
};

export const useUIStore = createStore<UIState>('UIStore', () => ({
  searchQuery: '',
  searchActive: false,
  languageMenuOpen: false,
  userMenuOpen: false,
}));
