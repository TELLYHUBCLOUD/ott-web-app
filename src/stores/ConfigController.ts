import { useConfigStore } from './ConfigStore';

import { API_HOST } from '#src/config';
import { IS_PROD_MODE } from '#src/utils/common';

const envQueryKey = 'env';

export const initializeHost = (searchParams: URLSearchParams) => {
  const query = searchParams.get(envQueryKey);

  if (IS_PROD_MODE) {
    return;
  }

  if (query === 'dev' || query === 'prd') {
    useConfigStore.setState({ apiHost: API_HOST[query] });
    return;
  }

  useConfigStore.setState({ apiHost: API_HOST.prd });
};
