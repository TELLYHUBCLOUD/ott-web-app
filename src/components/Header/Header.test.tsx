import React from 'react';

import { render } from '../../testUtils';
import { ConfigStore } from '../../stores/ConfigStore';
import Button from '../Button/Button';

import Header from './Header';

describe('<Header />', () => {
  test('renders header', () => {
    ConfigStore.update((s) => {
      s.config.sso = { host: 'https://www.aws.com', clientId: '12345CLIENT' };
    });

    const playlistMenuItems = [<Button key="key" label="Home" to="/" />];
    const { container } = render(
      <Header
        onMenuButtonClick={jest.fn()}
        searchBarProps={{
          query: '',
          onQueryChange: jest.fn(),
        }}
        searchEnabled
        searchActive={false}
        onSearchButtonClick={jest.fn()}
        onCloseSearchButtonClick={jest.fn()}
        onLoginButtonClick={jest.fn()}
        userMenuOpen={false}
        toggleUserMenu={jest.fn()}
        isLoggedIn={false}
        canLogin={true}
        showPaymentsMenuItem={true}
      >
        {playlistMenuItems}
      </Header>,
    );

    expect(container).toMatchSnapshot();
  });
});
