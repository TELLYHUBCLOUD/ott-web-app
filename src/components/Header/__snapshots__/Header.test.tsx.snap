// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders header 1`] = `
<div>
  <header
    class="header static"
  >
    <div
      class="container"
    >
      <div
        class="menu"
      >
        <div
          aria-label="open_menu"
          class="iconButton iconButton"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="icon"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </div>
      </div>
      <nav
        aria-label="menu"
        class="nav"
      />
      <div
        class="search"
      >
        <div
          class="searchBar"
        >
          <svg
            aria-hidden="true"
            class="icon icon"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </g>
          </svg>
          <input
            aria-label="search_bar.search_label"
            class="input"
            placeholder="search_bar.search_placeholder"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="buttonContainer"
      >
        <a
          class="button default outlined"
          href="https://www.aws.com/oauth2/authorize?client_id=12345CLIENT&response_type=code&redirect_uri=http://localhost/sso/login&state=eyJyZXR1cm5VcmwiOiIvIn0%3D"
        >
          <span>
            sign_in
          </span>
        </a>
        <a
          class="button primary"
          href="https://www.aws.com/signup?client_id=12345CLIENT&response_type=code&redirect_uri=http://localhost/sso/login&state=eyJyZXR1cm5VcmwiOiIvIn0%3D"
        >
          <span>
            sign_up
          </span>
        </a>
      </div>
    </div>
  </header>
</div>
`;
